lib/repeatable/expression/base.rb

```diff
@@ -24,6 +24,29 @@ def to_h
         )
       end
 
+      def union(other)
+        if other.is_a?(Union)
```

Comment:
> It might be better suited for some kind of "utilities" class to be opted into?

That was the other thought I had: if we need it, does it fit better somewhere else?  I think keeping it simple for now and keeping this solution in mind in the future is a good way forward.